pkg/debugapi/debugapi.go

```diff
@@ -103,28 +103,49 @@ func equalASCIIFold(s, t string) bool {
 	return s == t
 }
 
-func New(overlay swarm.Address, publicKey, pssPublicKey ecdsa.PublicKey, ethereumAddress common.Address, p2p p2p.DebugService, pingpong pingpong.Interface, topologyDriver topology.Driver, storer storage.Storer, logger logging.Logger, tracer *tracing.Tracer, tags *tags.Tags, accounting accounting.Interface, settlement settlement.Interface, chequebookEnabled bool, swap swap.ApiInterface, chequebook chequebook.Service, o Options) Service {
-	s := &server{
-		Overlay:           overlay,
-		PublicKey:         publicKey,
-		PSSPublicKey:      pssPublicKey,
-		EthereumAddress:   ethereumAddress,
-		P2P:               p2p,
-		Pingpong:          pingpong,
-		TopologyDriver:    topologyDriver,
-		Storer:            storer,
-		Logger:            logger,
-		Tracer:            tracer,
-		Tags:              tags,
-		Accounting:        accounting,
-		Settlement:        settlement,
-		metricsRegistry:   newMetricsRegistry(),
-		ChequebookEnabled: chequebookEnabled,
-		Chequebook:        chequebook,
-		Swap:              swap,
-	}
+// New creates a new Debug API Service with only basic routers enabled in order
+// to expose /health endpoint, Go metrics and pprof. It is useful to expose
+// these endpoints before all dependencies are configured and injected to have
+// access to basic debugging tools and /health endpoint.
+func New(logger logging.Logger, tracer *tracing.Tracer, corsAllowedOrigins []string) Service {
```

Comment:
Minor suggestion: I think that `overlay swarm.Address, publicKey, pssPublicKey ecdsa.PublicKey, ethereumAddress common.Address` can already be set in this constructor as they are static and not expected to change during runtime.

Reply:
That is a very good suggestion. The only problem is `ethereumAddress common.Address` which needs to wait for `InitChain` function to complete. I am suggesting to have explicit `SetEthereumAddress` method and move other address arguments to constructor as you suggested. The `EthereumAddress` would require lock, but that is not so drastic.

This way there would be a debug api available as soon as possible for debugging and health and also with /addresses endpoint exposed, but ethereum address would be set only if and after `InitChain` is done. This would address some of the points in https://github.com/ethersphere/bee/issues/1295 that @Eknir mentioned. WTYT @acud @Eknir?

Reply:
Actually, I don't think that you need `InitChain` for this. You can have the ethereum address from the signer which is passed to `NewBee`, exactly as it is used in `InitChain`

Reply:
Yes, correct. Great. :)